components/DropDown.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from 'react';
+import { StyleSheet, View, Dimensions } from 'react-native';
+import { Dropdown } from 'react-native-element-dropdown';
+
+interface DropdownComponentProps {
+  itemType: 'birthYear' | 'gender'; 
+}
+
+const windowWidth = Dimensions.get('window').width;
+
+const DropdownComponent: React.FC<DropdownComponentProps> = ({ itemType }) => {
+  const [selectedItem, setSelectedItem] = useState<string | null>(null);
+  const [isFocus, setIsFocus] = useState(false);
+  const [dropdownOptions, setDropdownOptions] = useState<Array<{ label: string; value: string }>>([]);
+
+  useEffect(() => {
+    if (itemType === 'birthYear') {
+      //  stores/generates the list of past 50 years
+      const currentYear = new Date().getFullYear();
+      const birthYear = Array.from({ length: 50 }, (_, index) => ({
+        label: (currentYear - index).toString(),
+        value: (currentYear - index).toString(),
+      }));
+      setDropdownOptions(birthYear);
+    } else if (itemType === 'gender') {
+      const genderOptions = [
+        { label: 'Male', value: 'male' },
+        { label: 'Female', value: 'female' },
+        { label: 'Other', value: 'other'},
+      ];
+      setDropdownOptions(genderOptions);
+    } 
+  }, [itemType]);
+
+  return (
+    <View>
+      <Dropdown
+        style={[styles.dropdown, isFocus && { borderColor: '#49108B' }]}
+        placeholderStyle={styles.placeholderStyle}
+        selectedTextStyle={styles.selectedTextStyle}
+        inputSearchStyle={styles.inputSearchStyle}
+        data={dropdownOptions}
+        search
+        maxHeight={300}
+        labelField="label"
+        valueField="value"
+        placeholder={!isFocus ? `Select ${itemType}` : '>'}
```

Comment:
the itemType here causes the placeholder to be weird, it instead says "Select gender" and "Select birthYear" instead of "Select Gender" and "Select Year of Birth", minor nitpick but please fix

Reply:
okay :))